base/src/main/java/io/spine/validate/RequiredField.java

```diff
@@ -0,0 +1,179 @@
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.validate;
+
+import com.google.common.base.Splitter;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.Descriptors;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+/**
+ * A message option that defines a combination of required fields for the message.
+ *
+ * The fields are separated with a {@code |} symbol, and combined with a {@code &} symbol.
+ *
+ * Example:
+ * <pre>
+ *     {@code
+ *     message PersonName {
+ *         option (required_field) = "given_name|honorific_prefix & family_name";
+ *         string honorific_prefix = 1;
+ *         string given_name = 2;
+ *         string middle_name = 3;
+ *     }
+ *     }
+ * </pre>
+ * The {@code PersonName} message is valid against the {@code RequiredField} either if it has a
+ * non-default family name, or both honorific prefix and a family name.
+ */
+public class RequiredField extends MessageValidatingOption<String> {
+
+    /**
+     * The name of the message option field.
+     */
+    private static final String OPTION_REQUIRED_FIELD = "required_field";
+
+    /**
+     * Combination of fields are made with ampersand.
+     */
+    private static final char AMPERSAND = '&';
+
+    private final ImmutableList.Builder<ConstraintViolation> violations = ImmutableList.builder();
+
+    @Override
+    List<ConstraintViolation> applyValidationRules(MessageValue message) {
+        String expression = getValueFor(message);
+        ImmutableList<RequiredFieldOptionData> parse = parse(expression);
+        if (!alternativeFound(parse, message)) {
+            String msgFormat =
+                    "None of the fields match the `required_field` definition: %s";
+            ConstraintViolation requiredFieldNotFound = ConstraintViolation
+                    .newBuilder()
+                    .setMsgFormat(msgFormat)
+                    .addParam(expression)
+                    .build();
+            violations.add(requiredFieldNotFound);
+        }
+        return violations.build();
+    }
+
+    @Override
+    boolean optionPresent(MessageValue message) {
+        return !getValueFor(message).isEmpty();
+    }
+
+    @Override
+    public String getValueFor(MessageValue message) {
+        Map<Descriptors.FieldDescriptor, Object> options = message.options();
+        for (Descriptors.FieldDescriptor optionDescriptor : options.keySet()) {
+            if (OPTION_REQUIRED_FIELD.equals(optionDescriptor.getName())) {
+                String expression = (String) options.get(optionDescriptor);
+                return expression;
+            }
+        }
+        return "";
+    }
+
+    private static class RequiredFieldOptionData {
```

Comment:
Please get rid of the unnecessary `Data` suffix. If a name not specific enough, re-consider the responsibility of the class.